apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.tapestry.pages;
+
+import org.apache.tapestry5.alerts.AlertManager;
+import org.apache.tapestry5.annotations.InjectComponent;
+import org.apache.tapestry5.annotations.Property;
+import org.apache.tapestry5.corelib.components.Form;
+import org.apache.tapestry5.ioc.annotations.Inject;
+import org.slf4j.Logger;
+
+public class Login {
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private AlertManager alertManager;
+
+    @InjectComponent
+    private Form login;
+
+    @Property
+    private String email;
+
+    @Property
+    private String password;
+
+    void onValidateFromLogin() {
```

Comment:
I think this should be onValidateFormLogin

Reply:
this is correct. The convention of the method name is on**Event**From**Component**